src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -264,21 +265,39 @@ object DefaultSource {
           .mkString(", ")}"))
   }
 
+  def parseAuth(parameters: Map[String, String]): Option[CdfSparkAuth] = {
+    val bearerToken = parameters.get("bearerToken").map(bearerToken => BearerTokenAuth(bearerToken))
+    val apiKey = parameters.get("apiKey").map(apiKey => ApiKeyAuth(apiKey))
+    val clientCredentials = for {
+      tokenUri <- parameters.get("tokenUri").map { tokenUriString =>
+        Uri
+          .parse(tokenUriString)
+          .getOrElse(throw new CdfSparkIllegalArgumentException("Invalid URI in tokenUri parameter"))
+      }
+      clientId <- parameters.get("clientId")
+      clientSecret <- parameters.get("clientSecret")
+      scopes <- parameters.get("scopes").map(_.split(" ").toList)
```

Comment:
Didn't think about it before now, but does the new auth setup require that people _have_ to list scopes every time they use their credentials? Is this to be able to grant a subset of your own access, or can it be left blank and still work?

Reply:
It needs to be specified, as far as I know. See https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#client-credentials-grant-flow-and-default

Reply:
This seems like a better explanation: https://dev.to/425show/just-what-is-the-default-scope-in-the-microsoft-identity-platform-azure-ad-2o4d

The "When must I use the /.default scope?" section in particular.

Reply:
So basically:
- We can't leave it blank
- We could automatically generate it based on the baseUrl, but:
  - It would only apply to azure
  - While the `{resource}` is a URL in our case, it does not have to be... I think

We will need to have some documentation on how to set this up for azure anyway (how to format the token endpoint URL, etc.), so I'm kinda leaning towards not doing too much nonstandard magic on behalf of the user, in exchange for the user having to do a bit more work.

Reply:
Since Azure would be most of our users, having a default generated based on "baseUrl" but allowing it to be overridden via "scopes" seems reasonable.